### PR TITLE
docs: split CONTRIBUTING.md into separate pages

### PR DESCRIPTION
**Pull request by lmb**
Split out and update CONTRIBUTING.md so it is part of the ebpf-go.dev documentation.

### CODE REVIEW

File: docs/ebpf/contributing/index.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+# How to contribute
+
+Development happens on [GitHub](https://github.com/cilium/ebpf) and contributions in
+all forms are welcome. Please take a look at [the architecture](architecture.md) to get
+a better understanding for the high-level goals.
```
Comment (ti-mo) [inline docs/ebpf/contributing/index.md:5]:
```suggestion
a better understanding of the high-level goals.
```

File: docs/ebpf/contributing/index.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+To test the current package with your local kernel you can simply run:
+```
+go test -exec sudo ./...
+```
+
+To test the current package with a different kernel version you can use the [run-tests.sh] script.
+It requires [virtme] and qemu to be installed.
+Unfortunately virtme is not well maintained at the moment, so we recommend installing
+a know working version:
```
Comment (ti-mo) [inline docs/ebpf/contributing/index.md:20]:
```suggestion
a known working version:
```

File: docs/ebpf/contributing/index.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+variables:
+
+```shell-session
+make CONTAINER_ENGINE=podman CONTAINER_RUN_ARGS=
+```
+
+### Updating kernel dependencies
+
+Syscall bindings and some parameters required to parse ELF sections are derived
+from upstream kernel version. You can update them to the latest version by:
```
Comment (ti-mo) [inline docs/ebpf/contributing/index.md:60]:
```suggestion
from upstream kernel versions. You can update them to the latest version by:
```

File: docs/ebpf/contributing/index.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+1. Adjusting the `KERNEL_VERSION` variable in `Makefile`
+2. Running
+    ```shell-session
+        make update-kernel-deps
```
Comment (ti-mo) [inline docs/ebpf/contributing/index.md:65]:
```suggestion
    make update-kernel-deps
```

File: docs/ebpf/contributing/index.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+    ```
+3. [Regenerating testdata and source code](#regenerating-testdata-and-source-code)
+
+## Project permissions
+
+If you'd like to contribute to the library more regularly, one of the
+[maintainers][ebpf-lib-maintainers] can add you to the appropriate team or mark
+you as a code owner. Just create an issue in the repository.
```
Comment (ti-mo) [inline docs/ebpf/contributing/index.md:73]:
```suggestion
you as a code owner. Please create an issue in the repository if you want to be involved more structurally in the maintenance of the project.
```

File: docs/ebpf/contributing/new-feature.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+   Also see [API stability](#api-stability).
+3. (*optional*) Create a draft PR if you want to discuss the implementation or have hit a problem. It's fine if this doesn't compile or contains debug statements.
+4. Create a PR that is ready to merge. This must pass CI and have tests.
+
+## API stability
+
+There is an emphasis on compatibility even though the library doesn't guarantee
+the stability of its API at the moment.
+
+1. If possible avoid breakage by introducing new API and deprecating the old one
```
Comment (ti-mo) [inline docs/ebpf/contributing/new-feature.md:23]:
```suggestion
1. If possible, avoid breakage by introducing new API and deprecating the old one
```

**Review by ti-mo — DISMISSED** [review]
Thanks! A few nits.

File: docs/ebpf/contributing/index.md (hunk 5 above)
Comment (lmb) [reply docs/ebpf/contributing/index.md:73]: Happy to change the `Please`, the other sentence doesn't add much for me. What's your intention?

**Pull request merged**